kafka-0.10.x/src/main/resources/monix/kafka/default.conf

```diff
@@ -77,4 +77,6 @@ kafka {
   monix.observable.commit.type = "sync"
   # Possible values: before-ack, after-ack or no-ack
   monix.observable.commit.order = "after-ack"
+  # Internal kafka heartbeat that avoids partition reasignment when the upstream subscribers is slow.
+  monix.observable.poll.heartbeat.rate.ms = 100
```

Comment:
Is there any recommended ratio between poll heartbeats and `max.poll.interval.ms`?

Reply:
@Avasil  in order to respond to that question, I have created benchmark with different poll intervals:

```
Benchmark                                               Mode  Cnt   Score   Error  Units
ConsumerBenchmark.monix_auto_commit10ms                thrpt   10  17.266 ± 2.231  ops/s
ConsumerBenchmark.monix_manual_commit_heartbeat1000ms  thrpt   10   2.971 ± 0.009  ops/s
ConsumerBenchmark.monix_manual_commit_heartbeat100ms   thrpt   10   9.477 ± 0.064  ops/s
ConsumerBenchmark.monix_manual_commit_heartbeat10ms    thrpt   10  14.710 ± 1.660  ops/s
ConsumerBenchmark.monix_manual_commit_heartbeat1ms     thrpt   10  15.494 ± 4.163  ops/s
```

Not sure why the performance is degraded with the increment of the `pollHeartBeat`, from the numbers, I would say that `10.ms` would be the best fit. I have made the property configurable only internally, since I think the user does not need to know about it. Any advice?

Reply:
I would expect that we could do this heartbeat very rarely. 
Is it only for manual commit?

I think the `commitAsync` can only complete after `poll` so that could be a potential cause? Perhaps we could do a heartbeat on commit as well? [Akka Streams](https://github.com/akka/alpakka-kafka/blob/master/core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala#L487) seems to be doing something like that but I have only had a quick glance and could be completely wrong.

> I have made the property configurable only internally, since I think the user does not need to know about it.

Is this the current state?

Reply:
Yes I did that after seeing that altering this value could impact substantially performance and the user should not be aware of it.
Ley me try your suggestion of polling after commit!